.github/workflows/version_check.yml

```diff
@@ -0,0 +1,198 @@
+              # Don't deserialise/serialise as this will scramble lines
+              lines = readlines(fname)
+              open(fname, "w") do io
+                  for line in lines
+                      if occursin(r"^Turing\s*=\s*\"\d+\.\d+\"\s*$", line)
+                          println(io, "Turing = \"$(target_version.:major).$(target_version.:minor)\"")
+                      else
+                          println(io, line)
+                      end
+                  end
+              end
+          end
+
+          function update_quarto_yml(fname, target_version)
+              # Don't deserialise/serialise as this will scramble lines
+              lines = readlines(fname)
+              open(fname, "w") do io
+                  for line in lines
+                      m = match(r"^(\s+)- text:\s*\"v\d+\.\d+\"\s*$", line)
+                      if m !== nothing
+                          println(io, "$(m[1])- text: \"v$(target_version.:major).$(target_version.:minor)\"")
+                      else
+                          println(io, line)
+                      end
+                  end
+              end
+          end
+
+          # Retain the original version number string for error messages, as
+          # VersionNumber() will tack on a patch version of 0
+          quarto_yaml = YAML.load_file(QUARTO_YML_PATH)
+          quarto_version_str = quarto_yaml["website"]["navbar"]["right"][1]["text"]
+          quarto_version = VersionNumber(quarto_version_str)
+          println("_quarto.yml version: ", quarto_version_str)
+
+          project_toml = TOML.parsefile(PROJECT_TOML_PATH)
+          project_version_str = project_toml["compat"]["Turing"]
+          project_version = VersionNumber(project_version_str)
+          println("Project.toml version: ", project_version_str)
+
+          manifest_toml = TOML.parsefile(MANIFEST_TOML_PATH)
+          manifest_version = VersionNumber(manifest_toml["deps"]["Turing"][1]["version"])
+          println("Manifest.toml version: ", manifest_version)
+
+          errors = []
+
+          if ENV["TARGET_IS_MASTER"] == "true"
+              # Fetch the latest version from GitHub and update files to match this version
+              # if necessary.
+
+              resp = HTTP.get("https://api.github.com/repos/TuringLang/Turing.jl/releases/latest")
+              latest_version = VersionNumber(JSON.parse(String(resp.body))["tag_name"])
+              println("Latest Turing.jl version: ", latest_version)
+
+              if !major_minor_match(latest_version, project_version)
+                  push!(errors, "$(PROJECT_TOML_PATH) out of date")
+                  println("$(PROJECT_TOML_PATH) is out of date; updating")
+                  update_project_toml(PROJECT_TOML_PATH, latest_version)
+              end
+
+              if !major_minor_match(latest_version, quarto_version)
+                  push!(errors, "$(QUARTO_YML_PATH) out of date")
+                  n_errors += 1
```

Comment:
This doesn't seem to be defined or used.

Reply:
Oops.